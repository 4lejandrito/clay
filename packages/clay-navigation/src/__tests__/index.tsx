/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayNavigation from '..';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';

describe('ClayNavigation', () => {
	afterEach(() => cleanup());

	const items = [
		{
			active: true,
			href: '#1',
			label: '1',
		},
		{
			href: '#2',
			label: '2',
		},
		{
			href: '#3',
			label: '3',
		},
		{
			href: '#4',
			label: '4',
		},
		{
			href: '#5',
			label: '5',
		},
	];

	it('renders', () => {
		const {container} = render(
			<ClayNavigation>
				<div />
			</ClayNavigation>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders a Breadcrumb', () => {
		const {container} = render(
			<ClayNavigation.Breadcrumb
				items={items}
				spritemap="path/to/spritemap"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('should throw a warning when not passing `active` to any `items` on Breadcrumb', () => {
		const mockWarnings = jest
			.spyOn(global.console, 'error')
			.mockImplementation(() => null);

		const items = [
			{
				href: '#1',
				label: '1',
			},
			{
				href: '#2',
				label: '2',
			},
		];

		const {container} = render(
			<ClayNavigation.Breadcrumb
				ellipsisBuffer={3}
				items={items}
				spritemap="path/to/spritemap"
			/>
		);

		expect(mockWarnings).toBeCalled();
		expect(mockWarnings.mock.calls[0][0]).toBe(
			'Warning: ClayNavigation.Breadcrumb expects at least one `active` item on `items`.'
		);

		expect(container).toMatchSnapshot();
		jest.resetAllMocks();
	});

	it('renders a Breadcrumb with properties passed by `ellipsisProps`', () => {
		const {container} = render(
			<ClayNavigation.Breadcrumb
				ellipsisBuffer={1}
				ellipsisProps={{style: {fontSize: '15px'}}}
				items={items}
				spritemap="path/to/spritemap"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('should emit an event when an item is clicked', () => {
		const itemClickMock = jest.fn(event => {
			event.persist();
		});

		const itemsWithoutHref = [
			{
				active: true,
				label: '1',
				onClick: itemClickMock,
			},
			{
				label: '2',
				onClick: itemClickMock,
			},
			{
				label: '3',
				onClick: itemClickMock,
			},
			{
				label: '4',
				onClick: itemClickMock,
			},
			{
				label: '5',
				onClick: itemClickMock,
			},
		];

		const {container, getByTestId} = render(
			<ClayNavigation.Breadcrumb
				ellipsisBuffer={1}
				items={itemsWithoutHref}
				spritemap="path/to/spritemap"
			/>
		);

		fireEvent.click(getByTestId('testId1'));

		expect(itemClickMock).toHaveBeenCalled();

		expect(itemClickMock.mock.calls[0][0].type).toBe('click');

		expect(container).toMatchSnapshot();
	});
});
