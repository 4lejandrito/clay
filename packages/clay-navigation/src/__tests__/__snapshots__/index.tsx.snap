// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayNavigation renders 1`] = `
<div>
  <div
    class=""
  >
    ClayNavigation
  </div>
</div>
`;

exports[`ClayNavigation renders a Breadcrumb 1`] = `
<div>
  <ol
    class="breadcrumb"
  >
    <li
      class="breadcrumb-item active"
    >
      <span
        class="breadcrumb-text-truncate"
        title="1"
      >
        1
      </span>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#2"
        role="button"
      >
        2
      </a>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#3"
        role="button"
      >
        3
      </a>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#4"
        role="button"
      >
        4
      </a>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#5"
        role="button"
      >
        5
      </a>
    </li>
  </ol>
</div>
`;

exports[`ClayNavigation renders a Breadcrumb with properties passed by \`ellipsisProps\` 1`] = `
<div>
  <ol
    class="breadcrumb"
  >
    <li
      class="breadcrumb-item active"
    >
      <span
        class="breadcrumb-text-truncate"
        title="1"
      >
        1
      </span>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#2"
        role="button"
      >
        2
      </a>
    </li>
    <li
      class="dropdown breadcrumb-item"
      style="font-size: 15px;"
    >
      <span
        style="display: inline-block;"
      >
        <button
          class="dropdown-toggle breadcrumb-link btn btn-unstyled"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-ellipsis-h"
            role="presentation"
          >
            <use
              xlink:href="path/to/spritemap#ellipsis-h"
            />
          </svg>
          <svg
            class="lexicon-icon lexicon-icon-caret-bottom"
            role="presentation"
          >
            <use
              xlink:href="path/to/spritemap#caret-bottom"
            />
          </svg>
        </button>
      </span>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#5"
        role="button"
      >
        5
      </a>
    </li>
  </ol>
</div>
`;

exports[`ClayNavigation should throw a warning when not passing \`active\` to any \`items\` on Breadcrumb 1`] = `
<div>
  <ol
    class="breadcrumb"
  >
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#1"
        role="button"
      >
        1
      </a>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="breadcrumb-link"
        href="#2"
        role="button"
      >
        2
      </a>
    </li>
  </ol>
</div>
`;
